source/fab/tasks/fortran.py

```diff
@@ -83,27 +88,29 @@ def run(self, hashed_file: HashedFile):
         log_or_dot(logger, f"analysing {fpath}")
 
         # parse the file
-        try:
-            tree = self._parse_file(fpath=fpath)
-        except Exception as err:
-            return err
-        if tree.content[0] is None:
+        parse_result = self._parse_file(fpath=fpath)
+        if isinstance(parse_result, Exception):
+            return parse_result
```

Comment:
### Short answer
The specific change you've highlighted here removes one of two nested try blocks: `_parse_file` already catches and returns any exception so there's no need to also do it in the caller, here.

### Discussion
There's a few places where we don't want vectorised multiprocessing to be aborted by an exception in a child process. In this example, a parsing error doesn't matter if the file is not subsequently required to build the target.

One pattern we use, as seen here, is for a child process to return a either a valid result or an exception. We then discard and summarise the errors, and carry on with the remaining results. This requires us to catch and return any exception in the child process. (There's too many broad catches in Fab, but that's another chat.)

We have a ticket to revisit exception handling in all the steps, including this. One idea is to use `map_async` instead of `map`. That would probably be an improvement, giving us simpler code, including letting child processes raise instead of return exceptions.